R/Spectra-functions.R

```diff
@@ -427,11 +428,14 @@ combineSpectra <- function(x, f = x$dataStorage, p = x$dataStorage,
     if (isReadOnly(x@backend))
         message("Backend of the input object is read-only, will change that",
                 " to an 'MsBackendDataFrame'")
-    ## We split the workload by storage file. This ensures memory efficiency
-    ## for file-based backends.
-    res <- bpmapply(FUN = .combine_spectra, split(x, p), split(f, p),
-                    MoreArgs = list(FUN = FUN, ...), BPPARAM = BPPARAM)
-    .concatenate_spectra(res)
+    if (length(levels(p)) > 1) {
```

Comment:
It is exactly the same code but I find it easier to read. You can ignore it if you like:
```suggestion
    if (nlevels(p)) > 1) {
```